lib/extendedkey.ex

```diff
@@ -5,44 +5,60 @@ defmodule Bitcoinex.ExtendedKey do
   alias Bitcoinex.Secp256k1.{Params, Point, PrivateKey, Math}
   alias Bitcoinex.Base58
 
+  use Bitwise, only_operators: true
+
   defmodule DerivationPath do
     @moduledoc """
     Contains a list of integers representing a bip32 derivation path
     """
 
+    @min_non_hardened_child_num 0
+    # 2^31 - 1
+    @max_non_hardened_child_num 0x80000000 - 1
+    @min_hardened_child_num 0x80000000
+    # 2^32 - 1
+    @max_hardened_child_num 0x100000000 - 1
```

Comment:
I think we can simplify this. We can use `>=` and `>` to avoid the need for variables for 2^32 and 2^32-1

Reply:
emm I am sure what's your suggestion here. previous I think hardcap, softcap is consusing as it is not so clear that if the cap value is inclusive or exclusive. I think these min, max naming is more clear?

Reply:
okay, this makes sense